ic-os/components/setupos-scripts/network.sh

```diff
@@ -168,45 +162,38 @@ function ping_ipv6_gateway() {
     echo " "
 }
 
-function assemble_nns_nodes_list() {
-    NNS_URL_STRING=$(/opt/ic/bin/fetch-property.sh --key=.nns.url --config=${DEPLOYMENT})
-    NNS_URL_LIST=$(echo $NNS_URL_STRING | sed 's@,@ @g')
-}
-
 function query_nns_nodes() {
     echo "* Querying NNS nodes..."
 
-    i=0
-    success=0
-    nodes=$(echo ${NNS_URL_LIST} | wc -w)
-    # At least one of the provided URLs needs to work.
-    verify=1
-    for url in $(echo $NNS_URL_LIST); do
+    nns_urls=$(get_config_value '.icos_settings.nns_urls')
+
+    local urls=($(echo ${nns_urls} | jq -r '.[]'))
+    local success=false
+
+    for url in "${urls[@]}"; do
         # When running against testnets, we need to ignore self signed certs
         # with `--insecure`. This check is only meant to confirm from SetupOS
         # that NNS urls are reachable, so we do not mind that it is "weak".
-        curl --insecure --head --connect-timeout 3 --silent ${url} >/dev/null 2>&1
-        if [ "${?}" -ne 0 ]; then
-            echo "  fail: ${url}"
-        else
+        if curl --insecure --head --connect-timeout 3 --silent "${url}" >/dev/null 2>&1; then
             echo "  okay: ${url}"
-            success=$((${success} + 1))
-        fi
-        i=$((${i} + 1))
-        if [ ${success} -ge ${verify} ]; then
-            echo "  success"
+            success=true
             break
-        elif [ ${i} -eq ${nodes} ]; then
-            log_and_halt_installation_on_error "1" "Unable to query enough healthy NNS nodes."
+        else
+            echo "  fail: ${url}"
         fi
     done
+
+    if $success; then
+        echo "  success"
+    else
+        log_and_halt_installation_on_error "1" "Unable to query enough healthy NNS nodes."
+    fi
 }
```

Comment:
I refactored this code to make it more readable and to handle the case where NNS_URL_STRING was empty (before, it wouldn't fail as the for loop as never triggered)